addons/mail/static/src/core/public_web/messaging_menu.js

```diff
@@ -119,6 +138,40 @@ export class MessagingMenu extends Component {
         return this.store.menuThreads;
     }
 
+    get standaloneInboxMsg() {
+        const tab = this.store.discuss.activeTab;
+        if (tab !== "main") {
+            return [];
+        }
+        return this.store.inbox.messages.filter(m => !m.thread);
+    }
+
+    get allMessagingItems() {
+        const threadMessages = this.threads
+            .map(thread => {
+                const message = (
+                    thread.isChatChannel ||
+                    (thread.channel_type === 'channel' && thread.needactionMessages.length === 0)
+                ) ? thread.newestPersistentOfAllMessage : thread.needactionMessages.at(-1);
+                return {
+                    thread,
+                    message,
+                    date: message?.datetime || 0,
+                };
+            });
+
+        const standaloneMessages = this.standaloneInboxMsg
+            .map(message => ({
+                thread: null,
+                message,
+                date: message?.datetime || 0,
+            }));
+
+        const allMsg = [...threadMessages, ...standaloneMessages];
+        allMsg.sort((a, b) => b.date - a.date);
+        return allMsg;
+    }
```

Comment:
I think this looks too complicated, too prone to bugs or performance issues to mix threads and messages with ordering and so on based on message datetime.
 
As a `[FIX]` I think just showing standalone inbox messages at the top is ok.
Note that we are aware of performance issues in messaging menu, which is linked to `sort` of `menuThreads`. having another sort will worsen the problem, so I would prefer to avoid adding another `sort` altogether with this fix, hence suggestion to just put them at the top.

FYI the task about the performance issue with menuThreads sort: https://www.odoo.com/odoo/1519/tasks/4794325